### PR TITLE
Minor changes and fixes

### PR DESCRIPTION
**Pull request by schustmi**
## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Describe changes
- Include AWS in list of integrations
- Remove old sphinx files
- Activate all integration when calling e.g. `zenml orchestrator up/down/logs`
- Only allow kwargs in `pipeline.run()`
- Use `BaseStep.entrypoint` instead of `getattr(BaseStep, STRING_CONSTANT)`

### CODE REVIEW

File: src/zenml/integrations/mlflow/mlflow_utils.py [hunk 1]
```diff
@@ -100,17 +100,18 @@ def enable_mlflow_run(run: Callable[..., Any]) -> Callable[..., Any]:
...
         This ensures each pipeline run is run within a different mlflow context.
 
         Args:
             self: self of the original pipeline class
             run_name: Optional name for the run.
+            **kwargs: Additional kwargs passed to `pipeline.run()`
         """
         with mlflow.start_run(run_name=run_name):
-            run(self, run_name)
+            return run(self, run_name=run_name, **kwargs)
```
Comment (AlexejPenner) [inline src/zenml/integrations/mlflow/mlflow_utils.py:114]: I am not too familiar with decorators - especially layerd like this - how come it worked before without returning the function? And why is it better here to return it?

**Review by AlexejPenner — APPROVED** [review]
Looks good, thanks for removing sphinx. I didn't wanna remove it in case mkdocs failed during the release, but now no more use for it :)

File: src/zenml/integrations/mlflow/mlflow_utils.py (hunk 1 above)
Comment (schustmi) [reply src/zenml/integrations/mlflow/mlflow_utils.py:114]: Essentially this makes sure the new `pipeline.run` method (the `inner_decorator`) returns the return value of the overwritten default `pipeline.run` method. In the local/kubeflow case this didn't matter as the method didn't return anything, but for airflow it returns an airflow object which gets forwarded here.

Comment (AlexejPenner) [reply src/zenml/integrations/mlflow/mlflow_utils.py:114]: Thanks for the explanation, makes sense

File: docs/book/features/integrations.md [hunk 2]
```diff
@@ -23,6 +23,7 @@ These are the third-party integrations that ZenML currently supports:
 | ----------- | ------ | ---- | -------------------- | ------- |
 | Apache Airflow | ✅ | Orchestrator | Works for local environment | [airflow_local](https://github.com/zenml-io/zenml/tree/main/examples/airflow_local) |
 | Apache Beam | ✅ | Distributed Processing |   |   |
+| AWS | ✅ | Cloud |   |   |
```
Comment (htahir1) [inline docs/book/features/integrations.md:26]: Maybe add a bit more detail here? :)

File: src/zenml/integrations/whylogs/whylogs_step_decorator.py [hunk 3]
```diff
@@ -97,7 +97,7 @@ def data_loader(
 
     def inner_decorator(_step: S) -> S:
 
-        source_fn = getattr(_step, STEP_INNER_FUNC_NAME)
+        source_fn = _step.entrypoint
```
Comment (htahir1) [inline src/zenml/integrations/whylogs/whylogs_step_decorator.py:100]: @stefannica might be worth conforming to this with the mlflow upgrade

**Review by htahir1 — COMMENTED** [review]
All looks good except a few minor nits!

File: docs/book/features/integrations.md (hunk 2 above)
Comment (alex-zenml) [reply docs/book/features/integrations.md:26]: We've really only added the ability to hook into S3 buckets for artifact stores and their filesystem, so we should be specific about that, I guess.

Comment (schustmi) [reply docs/book/features/integrations.md:26]: True, I was lazy and just copied the GCP line :D
I'll add a little description

Comment (schustmi) [reply docs/book/features/integrations.md]: done

**Review by htahir1 — APPROVED** [review]

**Review by alex-zenml — APPROVED** [review]

**Pull request merged**